### PR TITLE
[Typing Fixes]: Type fix + cleanup

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Issue
Type errors.

## Description
Solves the type errors and does some general cleanup.

### Double check

- [x] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

File: web/src/features/charts/bar-breakdown/utils.ts
Comment (VIKTORVAV99) [inline web/src/features/charts/bar-breakdown/utils.ts:34]: There is no storageCo2Intensities key in the zoneData.

File: web/src/features/charts/bar-breakdown/utils.ts
Comment (VIKTORVAV99) [inline web/src/features/charts/bar-breakdown/utils.ts:79]: This can potentially be null or undefined if we add the check here that sets it to 0 we avoid errors later.

File: web/src/features/charts/bar-breakdown/utils.ts
Comment (VIKTORVAV99) [inline web/src/features/charts/bar-breakdown/utils.ts:127]: Just renamed the argument for clarity.

**Comment by VIKTORVAV99** [comment]
This caused a error later down the road here:
image
Will keep it as a draft until that is fixed.

**Comment by VIKTORVAV99** [comment]
Tracked down the issue to the `MetricRatio` function, I'll add any potential fix in this PR.

**Comment by madsnedergaard** [comment]
Hey @VIKTORVAV99, it's nice that you did this - but we need to be better at aligning on who's doing what, as I also did this (but haven't pushed the PR yet) 😅  Will review and compare changes later today!

**Comment by VIKTORVAV99** [comment]
> Hey @VIKTORVAV99, it's nice that you did this - but we need to be better at aligning on who's doing what, as I also did this (but haven't pushed the PR yet) 😅  Will review and compare changes later today!

Oh 😅

Any chance you took care of the last error as well?

File: web/src/features/charts/bar-breakdown/utils.ts
Comment (madsnedergaard) [reply web/src/features/charts/bar-breakdown/utils.ts:62]: I made the function return 0 instead of handling it here :)

File: web/src/features/charts/bar-breakdown/utils.ts
Comment (madsnedergaard) [reply web/src/features/charts/bar-breakdown/utils.ts:34]: Yes, agree - but there also shouldn't be (see https://github.com/electricitymaps/electricitymaps-contrib/issues/3668)

**Review by madsnedergaard — COMMENTED** [review]
I merged my changes into this PR, hope that's okay :)

File: web/src/utils/helpers.ts
Comment (madsnedergaard) [inline web/src/utils/helpers.ts:40]: Made this change here, was waiting for feedback in https://github.com/electricitymaps/electricitymaps-contrib/issues/3668#issuecomment-1445826251 :)

**Review by madsnedergaard — APPROVED** [review]
Approving, but you should double check this too before merging :D

**Review by VIKTORVAV99 — COMMENTED** [review]
Can't approve my own PR but this LGTM! 🎉😅

**Pull request merged**
